dallinger/frontend/static/scripts/dallinger.js

```diff
@@ -237,16 +237,14 @@ var create_participant = function() {
 var lock = false;
 
 var submitResponses = function () {
-    submitNextResponse(0);
-    submitAssignment();
+    submitNextResponse(0, submitAssignment);
 };
 
 var submit_responses = function () {
     submitResponses();
-    submitAssignment();
 };
 
-var submitNextResponse = function (n) {
+var submitNextResponse = function (n, callback = function(){}) {
```

Comment:
I believe support for JS function parameter defaults is pretty limited in browsers. Probably best to simply define `function (n, callback) {...}` and check `if (callback !== undefined)` before calling it.

Reply:
Done